source/plugin/blocks/accessing.rst

```diff
@@ -34,56 +34,104 @@ referenced block is any kind of banner by checking the blocks type.
+
 Block States
 ~~~~~~~~~~~~
 
-Additional Data (like orientation, subtypes like granite vs regular stone, farm land moisture ... ) can be obtained from
-a ``BlockState``. A ``BlockState`` contains a ``BlockType`` and further data that can be accessed via a set of data
-manipulators that all extend the ``DataManipulator`` interface. The following example will return ``true`` if a the block
-at a given ``Location`` is a) a Sponge and b) wet, false otherwise.
+A ``BlockState`` contains a ``BlockType``,  any ``DataManipulator`` s and properties that are applied to the block, and any ``BlockTrait`` s for a block. It stores all immutable value's for a particular block. An use is of this is getting an  ``ImmutableDataManipulator``, as shown below:
 
 .. code-block:: java
 
     import org.spongepowered.api.block.BlockState;
-    import org.spongepowered.api.data.manipulator.WetData;
+    import org.spongepowered.api.data.manipulator.immutable.ImmutableWetData;
 
-    public boolean isWet(Location block) {
-        if (block.getBlock().getType() != BlockTypes.SPONGE) {
+    public void isWet(Location blockLoc) {
+        BlockState sponge = blockLoc.getBlock();
+        if (!sponge.getType().equals(BlockTypes.SPONGE)) {
             return false;
         }
-        BlockState sponge = block.getBlock();
-        Optional<WetData> wetness = sponge.getManipulator(WetData.class);
+        Optional<ImmutableWetData> wetness = sponge.get(ImmutableWetData.class);
         return wetness.isPresent();
     }
 
-First, we need to know which ``DataManipulator`` subinterface we need. Those that are applicable to blocks are found in
-the ``org.spongepowered.api.data.manipulator`` and ``org.spongepowered.api.data.manipulator.block`` packages. Then we
-can just pass that class to the ``getManipulator()`` method and get an ``Optional`` in return which will be ``empty()``
-if the ``BlockState`` does not contain data of that type. Since ``WetData`` represents a boolean value, its presence
-equates to ``true``. Its absence (if ``Optional.empty()`` was returned) either signifies ``false``.
+More information on mutable and immutable ``DataManipulator`` s can be found in the :doc:`data documentation <../data/datamanipulators>`.
+
+Block Properties
+~~~~~~~~~~~~~~~~
+Blocks can contain certain properties. A property is a pre-set value that defines the game logic of that particular block. For example, blocks can contain pre-determined blast-resistance values that can be used to determine what you're working with, without actually checking the type of block it could be one by one. For example, if we wanted to get the blast resistance of a block and checking if it is greater than or equal to one, it would be done like so:
```

Comment:
Please combine the first two sentences: "Blocks can contain certain properties, or preset values..."

There is also no hyphen in 'preset'